src/measure_theory/measure/with_density_vector_measure.lean

```diff
@@ -142,4 +142,39 @@ lemma integrable.with_densityᵥ_eq_iff {f g : α → E}
   μ.with_densityᵥ f = μ.with_densityᵥ g ↔ f =ᵐ[μ] g :=
 ⟨λ hfg, hf.ae_eq_of_with_densityᵥ_eq hg hfg, λ h, with_densityᵥ_eq.congr_ae h⟩
 
+section signed_measure
+
+lemma with_densityᵥ_sub_eq_with_density {f g : α → ℝ≥0∞}
+  (hfm : ae_measurable f μ) (hgm : ae_measurable g μ)
+  (hf : ∫⁻ x, f x ∂μ ≠ ∞) (hg : ∫⁻ x, g x ∂μ ≠ ∞) :
+  μ.with_densityᵥ (λ x, (f x).to_real - (g x).to_real) =
+  @to_signed_measure α _ (μ.with_density f)
+    (is_finite_measure_with_density (lt_top_iff_ne_top.2 hf)) -
+  @to_signed_measure α _ (μ.with_density g)
+    (is_finite_measure_with_density (lt_top_iff_ne_top.2 hg)) :=
```

Comment:
This is a weird lemma. It is simply some `sub_apply` lemmas to get to something of the form `F(a) - F(b) = G(a) - G(b)`, chained with a proof that `F(x) = G(x)` for all `x`. You should separate those two steps.
```
lemma with_densityᵥ_to_real {f : α → ℝ≥0∞} (hfm : ae_measurable f μ)
  (hf : ∫⁻ x, f x ∂μ ≠ ∞) :
  μ.with_densityᵥ (λ x, (f x).to_real) =
  @to_signed_measure α _ (μ.with_density f)
    (is_finite_measure_with_density (lt_top_iff_ne_top.2 hf)) :=
begin
  have hfi : integrable (λ x, (f x).to_real) μ := sorry,
  ext i hi,
  rw [with_densityᵥ_apply hfi hi, to_signed_measure_apply_measurable hi,
      with_density_apply _ hi, integral_to_real hfm.restrict],
  refine ae_lt_top' hfm.restrict (lt_of_le_of_lt _ (lt_top_iff_ne_top.2 hf)),
  conv_rhs { rw ← set_lintegral_univ },
  exact lintegral_mono_set (set.subset_univ _),
end

lemma with_densityᵥ_sub_eq_with_density {f g : α → ℝ≥0∞}
  (hfm : ae_measurable f μ) (hgm : ae_measurable g μ)
  (hf : ∫⁻ x, f x ∂μ ≠ ∞) (hg : ∫⁻ x, g x ∂μ ≠ ∞) :
  μ.with_densityᵥ (λ x, (f x).to_real - (g x).to_real) =
  @to_signed_measure α _ (μ.with_density f)
    (is_finite_measure_with_density (lt_top_iff_ne_top.2 hf)) -
  @to_signed_measure α _ (μ.with_density g)
    (is_finite_measure_with_density (lt_top_iff_ne_top.2 hg)) :=
begin
  have hfi : integrable (λ x, (f x).to_real) μ := sorry,
  have hgi : integrable (λ x, (g x).to_real) μ := sorry,
  ext i hi,
  rw [with_densityᵥ_sub' hfi hgi, with_densityᵥ_to_real hfm hf, with_densityᵥ_to_real hgm hg],
end
```
And I think that the second lemma is not worth having. We can simply use the one-line proof when needed.

Reply:
Thanks! I've added the primed version for `neg`, `add` and `smul` as well

Reply:
I think you should remove `with_densityᵥ_sub_eq_with_density`. It seems too specific to be widely useful, and has a very short proof.